clients/src/main/java/org/apache/kafka/common/requests/ProduceResponse.java

```diff
@@ -91,8 +103,9 @@ public ProduceResponse(Struct struct) {
                 int partition = partRespStruct.getInt(PARTITION_KEY_NAME);
                 short errorCode = partRespStruct.getShort(ERROR_CODE_KEY_NAME);
                 long offset = partRespStruct.getLong(BASE_OFFSET_KEY_NAME);
+                long timestamp = partRespStruct.getLong(TIMESTAMP_KEY_NAME);
```

Comment:
Could we add a comment before line 95 that we expect the caller to pass in a struct with the latest schema?